### DISCUSSION TITLE
Using with brandedoutcast/publish-nuget@v2?

### DISCUSSION THREAD

**Discussion by tig** [adamralph/minver#499] [Q&A] [answered]
I currently use `Directory.Build.props` to manually set release versions which get picked up by my `publish.yaml` to publish to nuget.

`brandedoutcast/publish-nuget@v2` has a `VERSION_FILE_PATH` that points to  `Directory.Build.props` with a regex:

```
          VERSION_FILE_PATH: Terminal.Gui/Directory.Build.props

          # Regex pattern to extract version info in a capturing group
          VERSION_REGEX: ^\s*<Version>(.*)<\/Version>\s*$
```

How do I modify my `publish.yaml` to pull the version info from the assembly since `Directory.build.props` no longer contains the version?

Or am I barking up the wrong tree?

**Comment by adamralph** [comment, marked as answer]
I'm not sure `brandedoutcast/publish-nuget` is well suited to work with MinVer.

`brandedoutcast/publish-nuget` seems to extract the version from a repository artifact (`VERSION_FILE_PATH`), build the package, and then tag the repo with the version. I.e. the workflow is: update the artifact->build the project->tag the repo.

MinVer works in the opposite way. It extracts the version from a tag, and provides it to the .NET SDK as an MSBuild property. I.e. the workflow is: tag the repo->build the project. Also, MinVer doesn't do any building or pushing to NuGet. It's purely a versioning package.

If you want to use MinVer, it may be better to drop `brandedoutcast/publish-nuget`. They seem like two incompatible approaches. I guess you may be able to get things working somehow using [`minver-cli`](https://www.nuget.org/packages/minver-cli) and writing the version into a temporary file, but it would be rather contrived. You'd also have to ensure that `brandedoutcast/publish-nuget` didn't attempt to tag the repo, since the tag would necessarily already have to exist in order for `minver-cli` to work. I would sooner recommend building your project using a regular `dotnet` invocation (e.g. `dotnet build`) and pushing your NuGet packages using a different mechanism. For example, you could run `dotnet nuget push` when the workflow was triggered by a tag.

**Comment by tig** [comment]
Thanks. Can you point me at a sample that uses `dotnet nuget push` "when the workflow was triggered by a tag"?

I have no deep need to stick with `publish-nuget`.

**Reply by adamralph** [reply]
IdentityServer does something like this. It's pushing to myget on every build is which triggered by a push to `main` or the creation of a tag, but you can change the condition to a tag only. (You can ignore the signing stuff, only the condition and the call to `dotnet nuget push` are relevant.)

https://github.com/DuendeSoftware/IdentityServer/blob/e5159f93db9ec263503c292323d07a02fe13ed03/.github/workflows/ci.yml#L42-L47
